apis/core.oam.dev/v1alpha2/core_types.go

```diff
@@ -349,9 +349,29 @@ type ComponentStatus struct {
 
 // Revision has name and revision number
 type Revision struct {
-	Name         string `json:"name"`
-	Revision     int64  `json:"revision"`
+	Name     string `json:"name"`
+	Revision int64  `json:"revision"`
+
+	// RevisionHash record the hash value of the spec of ApplicationConfiguration object.
+	// We're going to deprecate that by using ApplicationRevision
 	RevisionHash string `json:"revisionHash,omitempty"`
+
+	// AppRevisionHash record the detailed hash value of the spec of ApplicationRevision Object
+	// We should replace it with 'revisionHash' after the refactor finished
+	AppRevisionHash *AppRevisionHash `json:"appRevisionHash,omitempty"`
```

Comment:
Not sure why we need to have a complex hash structure that we can't compare.  We end up not even using it to compare.